python/lsst/qa/explorer/matchVisits.py

```diff
@@ -0,0 +1,122 @@
+"""
+"""
+import os
+import pandas as pd
+import numpy as np
+
+from lsst.pex.config import Config, Field
+from lsst.pipe.base import CmdLineTask, ArgumentParser
+from lsst.daf.persistence import NoResults
+from lsst.qa.explorer.match import match_lists
+from lsst.pipe.drivers.utils import TractDataIdContainer
+
+from .parquetTable import ParquetTable
+
+# Question: is there a way that LSST packages store data files?
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+class MatchVisitsConfig(Config):
+    coaddName = Field(dtype=str, default="deep", doc="Name of coadd")
+    matchRadius = Field(dtype=float, default=0.2, doc="match radius in arcseconds")
+
+
+class MatchVisitsTask(CmdLineTask):
+    """Write a tract-level table of closest-match visit IDs
+
+    Run this task on a tract, and it writes a full-tract `visitMatchTable` (a ParquetTable)
+    of coadd -> visit match ids and match distances, organized with a multi-level index.
+    """
+    _DefaultName = "matchVisits"
+    ConfigClass = MatchVisitsConfig
+
+    inputDataset = 'analysisCoaddTable_forced'
+    outputDataset = 'visitMatchTable'
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", cls.inputDataset,
+                               help="data ID, e.g. --id tract=12345",
+                               ContainerClass=TractDataIdContainer)
+        return parser
+
+    def matchCats(self, df1, df2, raColumn='coord_ra', decColumn='coord_dec'):
+        """Match two catalogs, represented as dataframes
+
+        This uses the `match_lists` function, that uses a KDTree for matching.
+
+        Parameters
+        ----------
+        df1, df2 : pandas.DataFrame
+            Catalogs to match
+
+        raColumn, decColumn : str
+            Names of ra and dec columns
+
+        Returns
+        -------
+        good : np.ndarray (bool)
+            Boolean array indicating which indices of df1 have valid matches
+
+        matchId : np.ndarray (dtype int)
+            Index of closest source in df2 to each source in df1.
+
+        distance : np.ndarray (dtype float)
+            Distance of match
+        """
+        ra1, dec1 = df1[raColumn].values, df1[decColumn].values
+        ra2, dec2 = df2[raColumn].values, df2[decColumn].values
+
+        dist, inds = match_lists(ra1, dec1, ra2, dec2, self.config.matchRadius/3600)
+
+        good = np.isfinite(dist)
```

Comment:
Are the dists returned for failed matches truly `inf` or are they `nan`? It helps to be explicit as to which value is expected.

Reply:
By experience I've found it can be either, which is why the `isfinite.` Not sure exactly why `match_lists` returns one or the other, but it's happened.